site/docs/types/type_system.md

```diff
@@ -0,0 +1,16 @@
+# Type System
+
+Substrait tries to cover the most common types used in data manipulation. To minimize type explosion, the project currently follows the guideline that a logical type should probably only be included in the specification if it is included in at least two of the following open source Projects: Apache Arrow, Apache Iceberg, Apache Spark and Trino. Types beyond this common core may be represented using [simple extensions](../extensions/index.md#simple-extensions).
+
+Substrait types fundamentally consist of four components:
+
+| Component                       | Condition           | Examples                                                  | Description
+| ------------------------------- | ------------------- | --------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+| [Class](type_classes.md)        | Always              | `i8`, `string`, `STRUCT`, extensions                      | Together with the parameter pack, describes the set of non-null values supported by the type. Subdivided into simple and compound type classes.
+| Nullability                     | Always              | Either `NULLABLE` (`?` suffix) or `REQUIRED` (no suffix)  | Describes whether values of this type can be null. Note that null is considered to be a special value of a nullable type, rather than the only value of a special null type.
+| [Variation](type_variations.md) | Always              | Either `[0]` (system-preferred) or an extension           | Allows different variations of the same type class to exist in a system at a time, usually distinguished by in-memory format.
```

Comment:
It would be good to note that i8 we frequently skip expressing the variation when writing about the default variation.

Reply:
Added "no suffix" to the example list, does that capture it? b70f3d9